spec/system/assign_topic_spec.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+describe "Assign | Assigning topics", type: :system, js: true do
+  let(:topic_page) { PageObjects::Pages::Topic.new }
+  let(:assign_modal) { PageObjects::Modals::Assign.new }
+  fab!(:staff_user) { Fabricate(:user, groups: [Group[:staff]]) }
+  fab!(:admin) { Fabricate(:admin) }
+  fab!(:topic) { Fabricate(:topic) }
+  fab!(:post) { Fabricate(:post, topic: topic) }
+
+  before do
+    SiteSetting.assign_enabled = true
+    sign_in(admin)
+  end
+
+  describe "with open topic" do
+    it "can assign and unassign" do
+      visit "/t/#{topic.id}"
+
+      topic_page.click_assign_topic
+      assign_modal.set_assignee(staff_user)
+      assign_modal.confirm
+
+      expect(find("#post_2")).to have_content("Assigned")
+      expect(find("#topic .assigned-to")).to have_content(staff_user.username)
+
+      topic_page.click_unassign_topic
+
+      expect(find("#post_3")).to have_content("Unassigned")
+      expect(page).not_to have_css("#topic .assigned-to")
```

Comment:
This one is dangerous. When you have this form it will never wait for it (2 seconds) to show before agreeing it's not on screen.

You should write it like this:

```
expect(page).to have_no_css("#topic .assigned-to")
```

This form will wait from 0 to 2s to declare it's not existing. If it's not existing from 0, then it will be satisfied right away.